### ISSUE TITLE
Feature Request: --min_queue_size

### ISSUE THREAD

**Issue by rmarable** [aws/aws-parallelcluster#1082]
This is a feature request for a setting called "--min_queue_size" that would force a ParallelCluster stack to always maintain "min_queue_size" compute instances alive at all times.  

As of today, there doesn't seem to be a way to deploy a cluster that starts out with 10 nodes, can scale out to a maximum of 50 nodes, but can "hold" itself to a minimum of one node so that users don't have to wait for new instances to spin up before jobs can begin executing during periods of low utilization.

**Comment by sean-smith** [comment]
Hi @rmarable,

Here's how you do that:

```ini
[cluster scal]
...
initial_queue_size = 10
maintain_initial_size = true
max_queue_size = 50
```

**Comment by rmarable** [comment]
Wouldn't this setting keep the cluster at 10 nodes with a maximum size of 50? I would like it to start out with 10 nodes, with a maximum of 50, but let it scale down to 1 node.

**Comment by sean-smith** [comment]
Gotcha, ya you can do that manually by setting the ASG desired from the CLI.

```ini
[cluster scale]
...
initial_queue_size = 1
maintain_initial_size = true
max_queue_size = 50
```

Then when the cluster starts, grab the asg name and do:
```bash
$ aws autoscaling set-desired-capacity --auto-scaling-group-name [name_from_console] --desired-capacity 10
```

**Comment by rmarable** [comment]
Thanks Sean - I will think about ways to automate this efficiently and get back to you.

**Comment by enrico-usai** [comment]
Hi @bioinformaticscro-aws ,

> When I try out this feature...

The issue is actually a feature request so the "feature" is not yet implemented.

> ...I get an error updating the stack

Do you mean are you executing the _pcluster update_ command?
If yes, I think your issue could be related to the "update" command and it is a different use case.

Could you please create a new issue by attaching your configuration and describing which parameters did you change during the update, etc. ?

Thank you

**Comment by demartinofra** [comment]
This feature has been addressed in version 2.9 where you can set a min_count for each compute_resource
